packages/material-ui-system/src/css.js

```diff
@@ -13,15 +13,32 @@ function omit(input, fields) {
   return output;
 }
 
+function handleCss(styleFunction, theme) {
+  const apply = styles => {
+    const cssStyles = omit(styles, styleFunction.filterProps);
+    const output = {
+      ...styleFunction({ theme, ...styles }),
+      ...cssStyles,
+    };
+
+    Object.keys(cssStyles).forEach(key => {
+      if (typeof cssStyles[key] === 'object') {
+        output[key] = apply(cssStyles[key]);
+      }
+    });
+
+    return output;
+  };
+
+  return apply;
+}
+
 function css(styleFunction) {
   const newStyleFunction = props => {
-    const output = styleFunction(props);
+    const output = styleFunction(props) || {};
```

Comment:
Why do we need to handle falsy values now?

Reply:
This is what I tried to explain above.

If output is `null` the following will also result in `null`:
```javascript
merge(output, styleFunction({ theme: props.theme, ...props.css })) // === null
```

And the test should fail, but it is saved because the `styleFunction.filterProps` is wrongly passed to the `omit` which results in the whole object being spread.

`styleFunction.filterProps` is an array:

```diff
- omit(props.css, [styleFunction.filterProps])
+ omit(props.css, styleFunction.filterProps)
```

Reply:
See: https://github.com/mui-org/material-ui/pull/19208